main/src/com/google/refine/expr/functions/math/RandomNumber.java

```diff
@@ -45,12 +45,28 @@
     
     @Override
     public Object call(Properties bindings, Object[] args) {
-        if (args.length == 2 && args[0] != null && args[0] instanceof Number
-                && args[1] != null && args[1] instanceof Number && ((Number) args[0]).intValue()<((Number) args[1]).intValue()) {
-            int randomNum = ThreadLocalRandom.current().nextInt(((Number) args[0]).intValue(), ((Number) args[1]).intValue()+1);
-            return randomNum;
+        if(args.length >= 2 && args[0] != null && args[0] instanceof Number
+                && args[1] != null && args[1] instanceof Number && ((Number) args[0]).intValue()<((Number) args[1]).intValue()){
+            int low = ((Number) args[0]).intValue();
+            int high = ((Number) args[1]).intValue();
+            int randomNum = ThreadLocalRandom.current().nextInt(low, high);
+            if(args.length == 2){
+                return randomNum;
+            } else if (args.length == 3){
+                Object o3 = args[2];
+                if (o3 != null && o3 instanceof String) {
+                    String type = ((String ) o3).toLowerCase();
+                    if(type == "long"){
+                        return (long)randomNum;
```

Comment:
Here we are casting an int to a long, so the random value cannot actually go beyond the range allowed by `int`, right? Intuitively we should generate a `long` in the first place, I think.